BoxContentSDK/BoxContentSDK/Models/BOXRepresentation.m

```diff
@@ -87,4 +87,36 @@ - (instancetype)initWithJSON:(NSDictionary *)JSONResponse
     return self;
 }
 
+- (id)initWithCoder:(NSCoder *)coder
+{
+    self = [super init];
+    if (self) {
+        _type = [coder decodeObjectOfClass:[BOXRepresentationType class] forKey:NSStringFromSelector(@selector(type))];
+        _status = [coder decodeObjectOfClass:[BOXRepresentationStatus class] forKey:NSStringFromSelector(@selector(status))];
+        _statusCode = [coder decodeObjectOfClass:[NSString class] forKey:NSStringFromSelector(@selector((statusCode)))];
+        _infoURL = [coder decodeObjectOfClass:[NSURL class] forKey:NSStringFromSelector(@selector(infoURL))];
+        _contentURL = [coder decodeObjectOfClass:[NSURL class] forKey:NSStringFromSelector(@selector(contentURL))];
+        _details = [coder decodeObjectOfClass:[NSDictionary class] forKey:NSStringFromSelector(@selector(details))];
+        _dimensions = [coder decodeObjectOfClass:[BOXRepresentationImageDimensions class] forKey:NSStringFromSelector(@selector(dimensions))];
+        _JSONData = [coder decodeObjectOfClass:[NSDictionary class] forKey:NSStringFromSelector(@selector(JSONData))];
```

Comment:
The dictionaries might need the element classes also added to be successful.

Reply:
don't think we can know for sure the subclasses in JSONDate or details dictionary. Without subclasses, that mean we can rely on Apple to decode to the best extend they can?

Reply:
I think we have to provide all the possibilities. We should test it. My guess is we want something like:

```
NSSet *jsonClasses =  [NSSet setFromArray:@[NSDictionary.class, NSArray.class, NSString.class, NSNumber.class, NSNull.class]];
[coder decodeObjectOfClasses:jsonClasses forKey:NSStringFromSelector(@selector(JSONData))];
```